### DISCUSSION TITLE
Intercepting HTTP requests with a DelegatingHandler

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#296] [Announcements]
# Intercepting HTTP requests with a DelegatingHandler

How and why you should use a DelegatingHandler to extract common HTTP request behavior.

https://timdeschryver.dev/blog/intercepting-http-requests-with-a-delegatinghandler

**Comment by klucyszy** [comment]
good article!

Please be careful on registering delegating handlers as scoped - the documentations states it should only be registered as transient ;) Just look in the summary of the `AddHttpMessageHandler` method and you will see that `The handler type must be registered as a transient service.`.

I would also add from my experience that you should avoid using complex logic inside DeletagingHandlers. We had complex AuthorizationHandler which had logic including EF queries and transactions (to load and store tokens) and it costs us a lot of pain. We need to resign from using DelegatingHandlers and inject that logic from outside service.

Best regards

**Reply by timdeschryver** [reply]
Thanks for sharing your experience and pointing that out @klucyszy .
